### ISSUE TITLE
[1.2.6] DataSourceLoader with requireTotalCount : true fails

### ISSUE THREAD

**Issue by core-avatar** [DevExpress/DevExtreme.AspNet.Data#120]
All working fine with 1.2.5, however when upgrading to 1.2.6 count queries fail.
In DataSourceLoader:100 
`return (int)source.Provider.Execute(builder.BuildCountExpr(source.Expression));`
throws something along the lines: `cannot convert IEnumerable<int> to IEnumerable<object>`

**Comment by AlekseyMartynov** [comment]
How can I reproduce it? Would you share a sample or a test case?

**Comment by core-avatar** [comment]
Key to failure is using an actual database
`    public class UnitTest1
    {

        public class DataSourceLoadOptions : DataSourceLoadOptionsBase
        {
        }

        public class Project
        {
            [Key]
            public long ID { get; set; }

            public string Name { get; set; }
        }

        public class ProjectContext : DbContext
        {
            protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
            {
                if (!optionsBuilder.IsConfigured)
                {
                    optionsBuilder.UseInMemoryDatabase(Guid.NewGuid().ToString("N"));
                }
            }

            public DbSet<Project> Projects { get; set; }
        }

        [Fact]
        public void ThisWorks()
        {
            using (var db = new ProjectContext())
            {
                DataSourceLoader.Load(db.Projects, new DataSourceLoadOptions
                {
                    RequireTotalCount = false
                });
            }
        }

        [Fact]
        public void ThisAlso()
        {
            var projects = new List<Project>
            {
                new Project { ID = 1, Name = "Test"}
            };

            DataSourceLoader.Load(projects, new DataSourceLoadOptions
            {
                RequireTotalCount = true
            });
        }

        [Fact]
        public void ThisFails()
        {
            using (var db = new ProjectContext())
            {
                DataSourceLoader.Load(db.Projects, new DataSourceLoadOptions
                {
                    RequireTotalCount = true
                });
            }
        }
    }`

**Comment by core-avatar** [comment]
I'd suggest integrating these tests into **DevExtreme.AspNet.Data.Tests.EF6**
or like I did use the EntityFrameworkCore.InMemory Provider.

**Comment by AlekseyMartynov** [comment]
Thanks.
Caused by https://github.com/aspnet/EntityFramework/issues/5982 manifested itself after #116

```c#
using(var db = new ProjectContext()) {
    var source = db.Projects.AsQueryable();
    var expr = Expression.Call(typeof(Queryable), "Count", new[] { typeof(Project) }, source.Expression);
    source.Provider.Execute(expr); // throws
}
```

Fixed in EF Core [2.0.0-preview1](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/2.0.0-preview1-final)

Since it's a confirmed fixed bug in EF Core, I'm not sure that we should work around it in the library code.
What do you think?

**Comment by core-avatar** [comment]
Well I can stay with 1.2.5 till hopefully EF Core 2.0 releases in a few months.

**Comment by AlekseyMartynov** [comment]
Good news. The generic `Execute<T>` doesn't fail.
The fix is on the way.

**Comment by AlekseyMartynov** [comment]
Closed in 39a38c4

**Comment by AlekseyMartynov** [comment]
FYI a new version (1.2.7) containing the fix has just been released.
